### PR TITLE
[BEAM-11456] Fix flaky table provider IT tests (Kafka and Pubsub)

### PR DESCRIPTION
**Pull request by piotr-szuberski**
Hopefully this will help with these tests being flaky or at least decrease flakiness rating at Java SQL PostCommit.

### CODE REVIEW

**Comment by piotr-szuberski** [comment]
Run SQL PostCommit

**Comment by piotr-szuberski** [comment]
Run SQL PostCommit

**Comment by piotr-szuberski** [comment]
Run SQL PostCommit

**Comment by piotr-szuberski** [comment]
Run SQL PostCommit

**Comment by piotr-szuberski** [comment]
Run SQL PostCommit

**Comment by piotr-szuberski** [comment]
Run SQL PostCommit

**Comment by piotr-szuberski** [comment]
Run SQL PostCommit

**Comment by piotr-szuberski** [comment]
Run SQL PostCommit

**Comment by piotr-szuberski** [comment]
R: @TheNeuralBit Could I ask you to review this small PR? You're the closest one to this part of code.

Kafka IT tests became flaky when they were running in parallel. It seems that setting 2 threads reduced the flakiness but only partially. Maybe when the tests are run on single thread they will stop being flaky. I also increased the timeout times, just in case.

File: sdks/java/extensions/sql/build.gradle [hunk 1]
```diff
@@ -177,16 +177,6 @@ task runPojoExample(type: JavaExec) {
...
-// These tests become flaky when run in parallel at more than 3 threads so run them in a separate task
-task runKafkaTableProviderIT(type: Test) {
-  outputs.upToDateWhen { false }
-  include '**/KafkaTableProvider*IT.class'
-  maxParallelForks 2
-  classpath = project(":sdks:java:extensions:sql").sourceSets.test.runtimeClasspath
-  testClassesDirs = files(project(":sdks:java:extensions:sql").sourceSets.test.output.classesDirs)
-  useJUnit { }
-}
-
```
Comment (TheNeuralBit) [inline sdks/java/extensions/sql/build.gradle:189]: Can we keep the Kafka test in a separate non-parallel task so the other ITs can still run in parallel?

**Review by TheNeuralBit — APPROVED** [review]
This looks good overall. I can merge if you address the build.gradle comment.

It would be better if we could address the root cause though. It seems likely that the issue is two tests running in parallel end up pointing to the same testcontainer, could you look into that? Maybe we need to find and specify a unique port for the bootstrap server in each test?

**Comment by piotr-szuberski** [comment]
Run SQL PostCommit

File: sdks/java/extensions/sql/build.gradle (hunk 1 above)
Comment (piotr-szuberski) [reply sdks/java/extensions/sql/build.gradle:189]: I'll replace the inheritance with parameterized tests, there shouldn't be need for a separate Kafka task. The same problem was with Pubsub IT tests so the task would have to receive GCP credentials as well to be able to run these as well.

**Comment by piotr-szuberski** [comment]
Run SQL PostCommit

**Comment by piotr-szuberski** [comment]
@TheNeuralBit Thanks for motivating me to do something better than reducing the thread amount.
I've got rid of inheritance and replaced it with parameterized tests and added some SuppressWarnings to make the code look cleaner in IntelliJ.

**Comment by piotr-szuberski** [comment]
Run SQL PostCommit

**Comment by aromanenko-dev** [comment]
Is it ready to be merged? I see that it was already approved.

**Comment by piotr-szuberski** [comment]
> Is it ready to be merged? I see that it was already approved.

Nope, there was an approval but I wanted to follow up the suggestion to do it in a better way and refactored the code so the tests now are parameterized and run sequentially. So it's waiting for review atm

**Comment by TheNeuralBit** [comment]
Sorry lost track of this! Taking a look now

**Review by TheNeuralBit — APPROVED** [review]
Re-structuring as a parameterized test makes a lot of sense, thank you!

**Pull request merged**